### PR TITLE
test: add comprehensive coverage for invalid investigation IDs with newline characters

### LINKED ISSUE #808: Add test for invalid investigation ID with newline character
## Problem
Invalid investigation IDs containing newline characters (e.g. `x\n`) are not tested, creating a potential edge case gap.

## Solution
Add test coverage for invalid IDs with newline characters in the remote server test suite at `tests/`.

## Use case
Ensures robustness of ID validation — path traversal and injection attacks are prevented even with unusual character sequences.

### PR DESCRIPTION
**Pull request by jerome-wilson**
Fixes #808 

#### Describe the changes you have made in this PR -

Added comprehensive test coverage for invalid investigation IDs containing newline characters and other injection attack vectors in `tests/remote/test_server_utils.py`.

**Changes:**
- Replaced the single `test_safe_investigation_path_rejects_x_newline` test with a parametrized test covering 11 injection vectors
- Test cases cover: LF (`\n`), CRLF (`\r\n`), CR (`\r`), null bytes (`\x00`), and Unicode line/paragraph separators (`\u2028`, `\u2029`)
- Ensures the `_safe_investigation_path()` function properly rejects all path traversal and injection attack attempts

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below) - but for verification.

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
Describe in your own words:
- What problem does your code solve?
- Solution: The existing test suite only covered a single newline case (`x\n`). This left potental edge case gaps for other injection attack vectors like CRLF, carriage returns, null bytes and Unicode line separators

- What alternative approaches did you consider?
- Solution: 
 % Adding individual test functions for each case - but it would create code duplicate
 % Using a simple list of test inputs - but it doesnt have a descriptive content

**Chosen implementation:** Used `pytest.mark.parametrize` with tuples of `invalid_id, description)` to create self-documenting test cases that are easy to maintain and extend.

- Why did you choose this specific implementation?
- What are the key functions/components and what do they do?
**Key components:**
- `@pytest.mark.parametrize` decorator: Generates 11 test cases from a single test function
- Each test case validates that `_safe_investigation_path()` raises `HTTPException` with status 400
- The `description` parameter documents the purpose of each test case for clarity

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

File: tests/remote/test_server_utils.py [hunk 1]
```diff
@@ -123,11 +123,36 @@ def test_safe_investigation_path_rejects_x_md() -> None:
...
+    Covers path traversal and injection attack vectors including:
+    - Newline characters (LF, CRLF, CR)
+    - Null bytes
+    - Unicode line/paragraph separators
+    """
     with pytest.raises(HTTPException) as exc_info:
-        _safe_investigation_path("x\n")
+        _safe_investigation_path(invalid_id)
     assert exc_info.value.status_code == 400
+    assert "Invalid investigation ID" in exc_info.value.detail
```
[the remark this answers is not in the document]
Comment (jerome-wilson) [reply tests/remote/test_server_utils.py]: attaches the descriptive ID directly to the test case correctly now

**Review by yashksaini-coder — APPROVED** [review]
Resolves a test coverage gap on `_safe_investigation_path()` — the old suite only checked a single `x\n` case. This expands it to 11 injection vectors (CRLF, null bytes, Unicode line separators, embedded/prefix positions) using `pytest.mark.parametrize`, and also tightens the assertion to verify the error message detail, not just the 400 status code. All 28 tests pass, lint and types are clean. Nice work.
